packages/block-library/src/query/utils.js

```diff
@@ -186,6 +188,60 @@ export function useIsPostTypeHierarchical( postType ) {
 	);
 }
 
+/**
+ * List of avaiable options to order by.
+ *
+ * @param {string} postType The post type to check.
+ * @return {OrderByOption[]} List of order options.
+ */
+export function useOrderByOptions( postType ) {
```

Comment:
This is a new helper method that returns all of the ordering options for a post type, accounting for the `page-attributes` support.

Reply:
Suggestion: We might want to return a memoized (`useMemo`) array of `options`; otherwise, it will break component memoization in the native files below.

https://github.com/WordPress/gutenberg/pull/68781/files#diff-095fb9f50f8c8d091aac13977a3b61bf1d439db2644bcf46d55d5cd7e459c582